c7n/resources/asg.py

```diff
@@ -553,6 +553,54 @@ def get_resource_date(self, i):
             self.date_attribute, "2000-01-01T01:01:01.000Z"))
 
 
+@filters.register('image')
+class ImageFilter(ValueFilter, LaunchConfigFilterBase):
+    """Filter asg by image
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: asg-image-tag
+                resource: asg
+                filters:
+                  - type: image
+                    value: "tag:ImageTag"
+                    key: "TagValue"
+                    op: eq
+    """
+    permissions = (
+        "ec2:DescribeImages",
+        "autoscaling:DescribeLaunchConfigurations")
+
+    schema = type_schema('image', rinherit=ValueFilter.schema)
+
+    def process(self, asgs, event=None):
+        self.initialize(asgs)
+        return super(ImageFilter, self).process(asgs, event)
+
+    def initialize(self, asgs):
+        super(ImageFilter, self).initialize(asgs)
+        image_ids = set()
+        for cfg in self.configs.values():
+            image_ids.add(cfg['ImageId'])
+        results = self.manager.get_resource_manager('ami').resources()
```

Comment:
by default the ami resources() method will filter to only images owned by the account. to pull in images from third parties that might be in use, this would need to use get_resources with the images ids, ie. say someone is using the ubuntu image published by canonical, it wouldn't show up in resources() but it would show up if using get_resources(). there might be a separate issue in doing that if the image has been deleted though.